modules/dicom_archive/jsx/dicom_archive.js

```diff
@@ -22,116 +22,158 @@ class DicomArchive extends Component {
+   *
+   * @return {object}
    */
   fetchData() {
-    $.ajax(this.props.DataURL, {
-      method: 'GET',
-      dataType: 'json',
-      success: function(data) {
-        loris.hiddenHeaders = data.hiddenHeaders ? data.hiddenHeaders : [];
-        this.setState({
-          Data: data,
-          isLoaded: true,
-        });
-      }.bind(this),
-      error: function(error) {
+    return fetch(this.props.dataURL, {credentials: 'same-origin'})
+      .then((resp) => resp.json())
+      .then((data) => this.setState({data}))
+      .catch((error) => {
+        this.setState({error: true});
         console.error(error);
-      },
-    });
+      });
   }
 
-  updateFilter(filter) {
-    this.setState({filter});
-  }
+  /**
+   * Modify behaviour of specified column cells in the Data Table component
+   *
+   * @param {string} column - column name
+   * @param {string} cell - cell content
+   * @param {object} row - row content indexed by column
+   *
+   * @return {*} a formated table cell for a given column
+   */
+  formatColumn(column, cell, row) {
+    let result = <td>{cell}</td>;
+    switch (column) {
+      case 'Archive Location': {
+        const downloadURL = '/mri/jiv/get_file.php?file=' + cell;
+        result =
+          <td>
+            <a href={downloadURL}>
+              <span className="glyphicon glyphicon-cloud-download"/>
+              &nbsp;
+              {cell}
+            </a>
+          </td>;
+      }
+      break;
+      case 'Metadata': {
+        const metadataURL = loris.BaseURL +
+          '/dicom_archive/viewDetails/?tarchiveID=' + row.TarchiveID;
+        result = <td><a href={metadataURL}>{cell}</a></td>;
+      }
+      break;
+      case 'MRI Browser': {
+        if (row.SessionID === null || row.SessionID === '') {
+          result = <td>&nbsp;</td>;
+        } else {
+          let mrlURL = loris.BaseURL + '/imaging_browser/viewSession/?sessionID=' +
+            row.SessionID;
+          result = <td><a href={mrlURL}>{cell}</a></td>;
+        }
+      break;
+      }
+      case 'INVALID - HIDDEN':
+        result = <td className="text-danger">{cell}</td>;
+      break;
+    }
 
-  resetFilters() {
-    this.refs.dicomFilter.clearFilter();
+    return result;
   }
 
   render() {
+    // If error occurs, return a message.
+    // XXX: Replace this with a UI component for 500 errors.
+    if (this.state.error) {
+      return <h3>An error occured while loading the page.</h3>;
+    }
+
     // Waiting for async data to load
     if (!this.state.isLoaded) {
-      return (
-        <button className="btn-info has-spinner">
-          Loading
-          <span
-            className="glyphicon glyphicon-refresh glyphicon-refresh-animate">
-          </span>
-        </button>
-      );
+      return <Loader/>;
     }
 
+    /**
+     * XXX: Currently, the order of these fields MUST match the order of the
+     * queried columns in _setupVariables() in media.class.inc
```

Comment:
```suggestion
     * queried columns in _setupVariables() in dicom_archive.class.inc
```

Xavier's comment as a suggestion